mmv1/third_party/terraform/tests/resource_billing_budget_test.go

```diff
@@ -283,3 +283,129 @@ func TestBillingBudgetStateUpgradeV0(t *testing.T) {
 		})
 	}
 }
+
+func TestAccBillingBudget_budgetFilterProjectsOrdering(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"org":             getTestOrgFromEnv(t),
+		"billing_acct":    getTestBillingAccountFromEnv(t),
+		"random_suffix_1": randString(t, 10),
+		"random_suffix_2": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckBillingBudgetDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBillingBudget_budgetFilterProjectsOrdering1(context),
+			},
+			{
+				ResourceName:      "google_billing_budget.budget",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+
+			{
+				Config:             testAccBillingBudget_budgetFilterProjectsOrdering2(context),
```

Comment:
I think we want another PlanOnly + ExpectNonEmptyPlan with the first config to make sure that both configs can apply without a plan